receiver/apachereceiver/config.go

```diff
@@ -29,6 +29,7 @@ type Config struct {
 	confighttp.HTTPClientSettings           `mapstructure:",squash"`
 	serverName                              string
 	Metrics                                 metadata.MetricsSettings `mapstructure:"metrics"`
+	ServerNameResourceAttrEnabled           bool                     `mapstructure:"server_name_resource_attr_enabled"`
```

Comment:
This should be controlled through a feature gate not a config option. Please look in this repo how it was applied in other parts. I remember at least docker stats receiver currently uses a feature gate